test/core.jl

```diff
@@ -5089,3 +5089,23 @@ m22929_2.x = m22929_1
 @test !isdefined_22929_x(m22929_1)
 @test isdefined_22929_1(m22929_2)
 @test isdefined_22929_x(m22929_2)
+
+# issue 18933
+module GlobalDef18933
+    using Base.Test
+    f() = (global sin; nothing)
```

Comment:
even worse, before we weren't even consistent about what this syntax did. for example:
```julia
if true
    global sin
    nothing
end
```
did not create a new global binding, but merely annotated that if a variable named `sin` was used, that its scope is global.